scripts/degenApe.js

```diff
@@ -115,301 +74,123 @@ const executeTx = async (calls, tx, fn, ...args) => {
-  constructor() {
-    super();
-  }
-}
-  `
-  const filePath = `${outputFolder}/implementations/${pfcoreArgs.chain}/${pfcoreArgs.chain}-${pfcoreArgs.protocols.join('-')}-${pfcoreArgs.extraTags.join('-')}-${pfcoreArgs.componentNames.join('-')}.ts`;
-
-  if (!fs.existsSync(filePath)) {
-    try {
-      fs.writeFileSync(filePath, output)
-    } catch (err) {
-      console.error(err)
-    }
-  }
-  console.log(output)
-
+  await sleep(sleepConfigs);
+  return transaction;
 }
 
 const deployContractsAndGeneratePfcore = async () => {
   for (const [jarIndex, contract] of contracts.entries()) {
     const StrategyFactory = await ethers.getContractFactory(contract);
     const PickleJarFactory = await ethers.getContractFactory("src/pickle-jar.sol:PickleJar");
     const Controller = await ethers.getContractAt("src/controller-v4.sol:ControllerV4", controller);
-    txRefs['name'] = contract.substring(contract.lastIndexOf(":") + 1);
+    txRefs.'name' = contract.substring(contract.lastIndexOf(":") + 1);
 
     try {
       // Deploy Strategy contract
-      console.log(`Deploying ${txRefs['name']}...`);
-      await executeTx(callAttempts, 'strategy', StrategyFactory.deploy.bind(StrategyFactory), governance, strategist, controller, timelock);
-      console.log(`✔️ Strategy deployed at: ${txRefs['strategy'].address} `);
+      console.log(`Deploying ${txRefs.'name'}...`);
+      txRefs.'strategy' = await executeTx(callAttempts, 'strategy', StrategyFactory.deploy.bind(StrategyFactory), governance, strategist, controller, timelock);
+      console.log(`✔️ Strategy deployed at: ${ txRefs.'strategy'.address } `);
 
       // Get Want
-      await sleep(sleepTime, sleepToggle);
-      txRefs['want'] = await txRefs['strategy'].want();
-
-      // Deploy PickleJar contract
-      await executeTx(callAttempts, 'jar', PickleJarFactory.deploy.bind(PickleJarFactory), txRefs['want'], governance, timelock, controller);
-      console.log(`✔️ PickleJar deployed at: ${txRefs['jar'].address} `);
+      await sleep(sleepConfigs);
+      txRefs.'want' = await txRefs.'strategy'.want();
 
       // Log Want
-      console.log(`Want address is: ${txRefs['want']} `);
-      console.log(`Approving want token for deposit...`);
-      await sleep(sleepTime, sleepToggle);
-      txRefs['wantContract'] = await ethers.getContractAt("ERC20", txRefs['want']);
-
-      // Approve Want
-      await executeTx(callAttempts, 'approveTx', txRefs['wantContract'].approve, txRefs['jar'].address, ethers.constants.MaxUint256);
-      console.log(`✔️ Successfully approved Jar to spend want`);
-      console.log(`Setting all the necessary stuff in controller...`);
+      console.log(`Want address is: ${ txRefs.'want' } `);
+      await sleep(sleepConfigs);
+      txRefs.'wantContract' = await ethers.getContractAt("ERC20", txRefs.'want');
+
+      // Check if Want already has a Jar on Controller
+      await sleep(sleepConfigs);
+      txRefs.'jar' = await Controller.jars(txRefs.'want');
```

Comment:
Putting everything in this txRefs object with no clear structure is beginning to become hard to follow at this point. Too many places are modifying it and the scope and domain seem like, well, like you don't need a single object holding all this stuff. It could all just be stored in normal defined local variables.